.github/workflows/format-native.yml

```diff
@@ -24,3 +24,20 @@ jobs:
     - name: Format native code
       shell: bash
       run: ./scripts/format-native.sh
+
+  check-native-headers:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v3.3.0
+      - uses: actions/setup-go@v3.5.0
+        with:
+          go-version: '1.20.1'
+      - name: Verify all native files have license headers
+        shell: bash
+        run: |
```

Comment:
Can you check if `run` is not retruning only the status of the last command?
I mean I think that if `go install github.com/google/addlicense@v1.1.1` fails and `test -z "$(git status --porcelain)"` passed then probably the job will be marked as passed.

How about moving this code to a `.sh` script like I have done here: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/.github/workflows/format-native.yml

It would also make it easier for local usage.

Reply:
> Can you check if `run` is not retruning only the status of the last command?
> I mean I think that if `go install github.com/google/addlicense@v1.1.1` fails and `test -z "$(git status --porcelain)"` passed then probably the job will be marked as passed.

My testing on fork indicates that failure for any command will fail the step and the job.

> How about moving this code to a `.sh` script like I have done here: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/.github/workflows/format-native.yml
> 
> It would also make it easier for local usage.

Sounds good, let me know if you'd like to do that before merging.

Reply:
> Can you check if run is not returning only the status of the last command?

If fails the step at the first failing command.

I would still prefer to have it as a script, but maybe it would make more sense to refactor to a Nuke target later... For now, I am OK to keep it as it is.